integrations/oci/tls-certificates/src/main/java/io/helidon/integrations/oci/tls/certificates/DefaultOciCertificatesDownloader.java

```diff
@@ -0,0 +1,122 @@
+import com.oracle.bmc.certificates.responses.GetCertificateAuthorityBundleResponse;
+import com.oracle.bmc.certificates.responses.GetCertificateBundleResponse;
+import jakarta.inject.Singleton;
+
+import static io.helidon.integrations.oci.tls.certificates.spi.OciCertificatesDownloader.create;
+
+/**
+ * Implementation of the {@link OciCertificatesDownloader} that will use OCI's Certificates Service to download certs.
+ */
+@Singleton
+class DefaultOciCertificatesDownloader implements OciCertificatesDownloader {
+
+    @Override
+    public Certificates loadCertificates(String certOcid) {
+        Objects.requireNonNull(certOcid);
+        try {
+            return loadCerts(certOcid);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to load certificate ocid: " + certOcid, e);
+        }
+    }
+
+    @Override
+    public Certificate loadCACertificate(String caCertOcid) {
+        Objects.requireNonNull(caCertOcid);
+        try {
+            return loadCACert(caCertOcid);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to load ca certificate ocid: " + caCertOcid, e);
+        }
+    }
+
+    static Certificates loadCerts(String certOcid) throws Exception {
+        try (CertificatesClient client = CertificatesClient.builder()
+                .build(OciExtension.ociAuthenticationProvider().get())) {
+            GetCertificateBundleResponse res =
+                    client.getCertificateBundle(GetCertificateBundleRequest.builder()
+                                                        .certificateId(certOcid)
+                                                        .build());
+            ByteArrayInputStream chainIs = new ByteArrayInputStream(res.getCertificateBundle().getCertChainPem()
+                                                                            .getBytes(StandardCharsets.US_ASCII));
+            ByteArrayInputStream certIs = new ByteArrayInputStream(res.getCertificateBundle().getCertificatePem()
+                                                                           .getBytes(StandardCharsets.US_ASCII));
+            Certificate[] certs = toCertificates(chainIs, certIs);
+            String version = toVersion(res.getEtag(), certs);
+            return create(version, certs);
+        }
+    }
+
+    static Certificate loadCACert(String caCertOcid) throws Exception {
+        GetCertificateAuthorityBundleResponse res;
+        try (CertificatesClient client = CertificatesClient.builder()
+                .build(OciExtension.ociAuthenticationProvider().get())) {
+            res = client.getCertificateAuthorityBundle(GetCertificateAuthorityBundleRequest.builder()
+                                                               .certificateAuthorityId(caCertOcid)
+                                                               .build());
+
+            byte[] pemBytes = res.getCertificateAuthorityBundle().getCertificatePem()
```

Comment:
This looks familiar - please check `PemReader` - if this does the same thing, then please expose this as public from our `pki` module and re-use

Reply:
pushed in last commit